### ISSUE TITLE
Web [Charts]: Improve Visx `<Axis/>` API in order to simplify props passing to the tick components

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#39874]
This issiue is a follow up for https://github.com/sourcegraph/sourcegraph/pull/39805. 

## Background 

Curretly for chart ticks aria-labels we have to use [`tickLabelProps`](https://airbnb.io/visx/docs/axis#Axis_tickLabelProps) prop of VisxAxis component. But in the same time we use [`tickComponent`](https://airbnb.io/visx/docs/axis#Axis_tickComponent) props for custom tick rendering (also because of a11y reasons). It would be better to use only one solution for tick props generation. Unfortanatly `tickComponent` doesn't have all needed data that we have inside `tickLabelProps` callback. 

In this issue we should contribute to `@visx/axis` package and extend axis component API (`TickRendererProps` props interface) with all needed data for tick aria-labels attributes (tick raw value, list of all ticks in axis).

**Comment by sourcegraph-bot-2** [comment]
Heads up @joelkw @felixfbecker @vovakulikov @unclejustin - the "team/code-insights" label was applied to this issue.
